File: reference-apps/nextjs-postgres-crud/prisma/seed.mjs
// Dev/test-only seed data. Passwords below are intentionally public defaults for a local
// database that only ever exists inside docker-compose.yml — never reuse them anywhere real.
import { PrismaClient } from "@prisma/client";
import { PrismaPg } from "@prisma/adapter-pg";
import { randomBytes, scrypt } from "node:crypto";
import { promisify } from "node:util";

const scryptAsync = promisify(scrypt);
const connectionString = process.env.DATABASE_URL;
if (!connectionString) throw new Error("DATABASE_URL is not set — run with `node --env-file=.env prisma/seed.mjs`");
const prisma = new PrismaClient({ adapter: new PrismaPg({ connectionString }) });

// Duplicated from lib/auth.ts on purpose: this script runs as plain Node ESM (no Next.js/TS
// build step involved), so it can't import the app's TypeScript module directly.
async function hashPassword(password) {
  const salt = randomBytes(16).toString("hex");
  const derived = await scryptAsync(password, salt, 64);
  return `${salt}:${derived.toString("hex")}`;
}

async function main() {
  const admin = await prisma.user.upsert({
    where: { email: "admin@example.com" },
    update: {},
    create: {
      email: "admin@example.com",
      name: "Admin",
      role: "admin",
      passwordHash: await hashPassword("admin-dev-password"),
    },
  });

  const member = await prisma.user.upsert({
    where: { email: "member@example.com" },
    update: {},
    create: {
      email: "member@example.com",
      name: "Member",
      role: "member",
      passwordHash: await hashPassword("member-dev-password"),
    },
  });

  await prisma.task.upsert({
    where: { id: "seed-task-1" },
    update: {},
    create: {
      id: "seed-task-1",
      title: "Write the runbook",
      status: "todo",
      ownerId: member.id,
    },
  });

  console.log(`Seeded: admin=${admin.email} member=${member.email}`);
}

main()
  .catch((error) => {
    console.error(error);
    process.exitCode = 1;
  })
  .finally(async () => {
    await prisma.$disconnect();
  });

File: reference-apps/expo-fastapi-postgres-sync/backend/scripts/seed.py
"""Dev/test-only seed data. Passwords below are intentionally public defaults for a local
database that only ever exists inside docker-compose.yml — never reuse them anywhere real.
"""

from sqlalchemy import select

from app.auth import hash_password
from app.db import SessionLocal
from app.models import Task, User


def main() -> None:
    db = SessionLocal()
    try:
        admin = db.execute(select(User).where(User.email == "admin@example.com")).scalar_one_or_none()
        if admin is None:
            admin = User(
                email="admin@example.com",
                name="Admin",
                role="admin",
                password_hash=hash_password("admin-dev-password"),
            )
            db.add(admin)
            db.flush()

        member = db.execute(select(User).where(User.email == "member@example.com")).scalar_one_or_none()
        if member is None:
            member = User(
                email="member@example.com",
                name="Member",
                role="member",
                password_hash=hash_password("member-dev-password"),
            )
            db.add(member)
            db.flush()

        seed_task = db.execute(select(Task).where(Task.id == "seed-task-1")).scalar_one_or_none()
        if seed_task is None:
            db.add(Task(id="seed-task-1", title="Write the sync contract doc", status="todo", owner_id=member.id))

        db.commit()
        print(f"Seeded: admin={admin.email} member={member.email}")
    finally:
        db.close()


if __name__ == "__main__":
    main()
